odoo/addons/base/models/qweb.py

```diff
@@ -1673,17 +1731,20 @@ def _get_widget(self, value, expression, tagName, field_options, options, values
     # compile expression
 
     def _compile_strexpr(self, expr):
-        # ensure result is unicode
         return ast.Call(
-            func=ast.Name(id='to_text', ctx=ast.Load()),
-            args=[self._compile_expr(expr)], keywords=[],
-            starargs=None, kwargs=None
+            func=ast.Name(id='str', ctx=ast.Load()),
+            args=[
+                ast.Call(
+                    func=ast.Name(id='to_text', ctx=ast.Load()),
+                    args=[self._compile_expr(expr)], keywords=[],
+                )
+            ], keywords=[]
```

Comment:
Note: JS behaves slightly differently here, non-text values can pass through as-is and are only stringified (with null/undefined being replaced by empty strings) in the `_append_attribute` equivalent. This is for the text-based compiler, I've not looked at the other one.